.github/workflows/ci.yml

```
@@ -1,46 +1,69 @@
-# This file is autogenerated by the 'modulegen' tool.
-# Please update the 'ci.yml' template instead.
 name: Main pipeline
 
 on:
   push:
     branches:
       - main
-    paths-ignore:
-      - '.vscode/**'
-      - 'mkdocs.yml'
-      - 'docs/**'
-      - 'README.md'
   pull_request:
-    paths-ignore:
-      - '.vscode/**'
-      - 'mkdocs.yml'
-      - 'docs/**'
-      - 'README.md'
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.sha }}
   cancel-in-progress: true
 
 jobs:
+  detect-modules:
+    runs-on: ubuntu-latest
+    outputs:
+      modules: ${{ steps.set-modified-modules.outputs.modules }}
+      modules_count: ${{ steps.set-modified-modules-count.outputs.modules_count }}
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - id: changed-files
+        name: Get changed files
+        uses: tj-actions/changed-files@v45
```

What do you think to pinning the commit? 

```suggestion
        uses: tj-actions/changed-files@4edd678ac3f81e2dc578756871e4d00c19191daf # v45.0.4
```

---

Done, thanks for this! 🙏